### PR TITLE
Add Neva template 

### PR DESCRIPTION
**Pull request by HuiyingLi**
# What does this PR do ?

Add template for DPO model training in NeVA 

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/multimodal/data/neva/neva_dataset.py [hunk 1]
```diff
@@ -462,6 +462,109 @@ def preprocess_nvgpt(sources: dict, tokenizer, cfg,) -> Dict:
...
+        - 'tokens': A tensor of tokenized conversation data.
+        - 'labels': A tensor of labels for the conversation data, used for training models. Labels are masked based on the conversation structure.
+
+    Note:
+    - The function includes specific token replacements (e.g., DEFAULT_IMAGE_PATCH_TOKEN, <s>, </s>) and masking techniques for labels.
+    - It is designed to work with conversational data where messages alternate between a 'human' and a 'gpt' role.
+    - The function asserts that each message in a conversation alternates between the defined roles and skips messages not starting with the 'human' role.
+    """
+
+    """<extra_id_0>System\n\n<extra_id_1>User\n{user input}\n<extra_id_1>Assistant\n"""
```
Comment (yaoyu-33) [inline nemo/collections/multimodal/data/neva/neva_dataset.py:487]: Can you help me add this template text to preprocess nvgpt function as well?

File: nemo/collections/nlp/modules/common/text_generation_utils.py [hunk 2]
```diff
@@ -181,14 +181,16 @@ def megatron_neva_generate(model, prompt_dict_list, length_params, sampling_para
...
+        if conv_template in ["nvgpt", "nv_steerlm"]:
             labels_str_regexp = re.compile(f"<extra_id_2>quality:.*\n")
             last_match_end_position = None
             for match in re.finditer(labels_str_regexp, clean_response):
                 last_match_end_position = match.end()
             if last_match_end_position is not None:
                 clean_response = clean_response[last_match_end_position:]
             clean_response = clean_response.strip("<extra_id_1>")
+        elif conv_template == 'nv_dpo':
+            clean_response = clean_response.split("<extra_id_1>")[-2].strip().split("\n")[-1]
```
Comment (yaoyu-33) [inline nemo/collections/nlp/modules/common/text_generation_utils.py:193]: `.strip().split("\n")[-1]` need all these? there isa strip on line 199 below

Comment (HuiyingLi) [reply nemo/collections/nlp/modules/common/text_generation_utils.py:193]: `clean_response.split("<extra_id_1>")[-2]` would give "Assistant\n{GENERATION}\n", then use `.strip().split("\n")[-1]` to grab the {GENERATION}. If directly use line 199 "Assistant" remains if I understand correctly

**Comment by yaoyu-33** [comment]
jenkins

**Review by yaoyu-33 — APPROVED** [review]

**Pull request merged**